Test can fail on shutdown due to netty thread blocking

### Describe the bug

We can get a blocking issue when netty is shutting down in some cases. We should ignore this in our blockhound exclusions.
```
08:22:12,267 ERROR (non-blocking-thread-MemcachedBinaryAuthenticationTest-NodeA-p15-t4:[]) [o.i.c.t.TestSuiteProgress] Test failed: org.infinispan.server.memcached.binary.MemcachedBinaryAuthenticationTest.BlockingChecker
java.lang.AssertionError: Blocking call! java.lang.Thread.sleep on thread Thread[non-blocking-thread-MemcachedBinaryAuthenticationTest-NodeA-p15-t4,5,ISPN-non-blocking-group]
	at org.infinispan.util.CoreTestBlockHoundIntegration.lambda$applyTo$0(CoreTestBlockHoundIntegration.java:51) ~[infinispan-core-15.0.15.Final-temporary-redhat-00003-tests.jar:15.0.15.Final-temporary-redhat-00003]
	at reactor.blockhound.BlockHound$Builder.lambda$install$8(BlockHound.java:488) ~[blockhound-1.0.10.RELEASE.jar:?]
	at reactor.blockhound.BlockHoundRuntime.checkBlocking(BlockHoundRuntime.java:89) [?:?]
	at java.base/java.lang.Thread.sleep(Thread.java) [?:?]
	at io.netty.util.concurrent.SingleThreadEventExecutor.confirmShutdown(SingleThreadEventExecutor.java:790) [netty-common-4.1.119.Final-redhat-00004.jar:4.1.119.Final-redhat-00004]
	at io.netty.channel.epoll.EpollEventLoop.run(EpollEventLoop.java:423) [netty-transport-classes-epoll-4.1.119.Final-redhat-00004.jar:4.1.119.Final-redhat-00004]
	at io.netty.util.concurrent.SingleThreadEventExecutor$4.run(SingleThreadEventExecutor.java:998) [netty-common-4.1.119.Final-redhat-00004.jar:4.1.119.Final-redhat-00004]
	at io.netty.util.internal.ThreadExecutorMap$2.run(ThreadExecutorMap.java:74) [netty-common-4.1.119.Final-redhat-00004.jar:4.1.119.Final-redhat-00004]
	at java.base/java.lang.Thread.run(Thread.java:840) [?:?]
```

### Expected behavior

_No response_

### Actual behavior

_No response_

### How to Reproduce?

_No response_

### Output of `uname -a` or `ver`

_No response_

### Output of `java -version`

_No response_

### Infinispan version or git rev

_No response_

### Build tool (ie. output of `mvnw --version` or `gradlew --version`)

_No response_

### Additional information

_No response_

--- a/server/core/src/main/java/org/infinispan/server/core/ServerCoreBlockHoundIntegration.java
+++ b/server/core/src/main/java/org/infinispan/server/core/ServerCoreBlockHoundIntegration.java
@@ -5,6 +5,7 @@ import java.util.Arrays;
 import org.infinispan.server.core.utils.SslUtils;
 import org.kohsuke.MetaInfServices;
 
+import io.netty.util.concurrent.SingleThreadEventExecutor;
 import reactor.blockhound.BlockHound;
 import reactor.blockhound.BlockingOperationError;
 import reactor.blockhound.integration.BlockHoundIntegration;
@@ -22,6 +23,9 @@ public class ServerCoreBlockHoundIntegration implements BlockHoundIntegration {
 
       // Nashorn prints to stderr in its constructor
       builder.allowBlockingCallsInside("jdk.nashorn.api.scripting.NashornScriptEngineFactory", "getScriptEngine");
+
+      // Don't worry about blocking in netty event loop when shutting down
+      builder.allowBlockingCallsInside(SingleThreadEventExecutor.class.getName(), "confirmShutdown");
 
       questionableBlockingMethod(builder);
       methodsToBeRemoved(builder);

